meson.build

```diff
@@ -1460,8 +1460,17 @@ videotoolbox_gl = get_option('videotoolbox-gl').require(
     error_message: 'gl-cocoa nor ios-gl could be found!',
 )
 features += {'videotoolbox-gl': videotoolbox_gl.allowed()}
+corevideo = dependency('appleframeworks', modules: ['CoreVideo'], required: get_option('videotoolbox-pl'))
+videotoolbox_pl = get_option('videotoolbox-pl').require(
+    features['vulkan'] and corevideo.found(),
+    error_message: 'vulkan or CV metal support could be found!',
+)
+features += {'videotoolbox-pl': videotoolbox_pl.allowed()}
 if features['videotoolbox-gl']
-    sources += files('video/out/opengl/hwdec_osx.c')
+    sources += files('video/out/hwdec/hwdec_mac_gl.c', 'video/out/hwdec/hwdec_vt.c')
+endif
+if features['videotoolbox-pl']
+    sources += files('video/out/hwdec/hwdec_vt_pl.m')
```

Comment:
Missing video/out/hwdec/hwdec_vt.c

Reply:
Maybe pull it out in a separate check completely like:
```
if features['videotoolbox-gl'] or features['videotoolbox-pl']
    sources += 'video/out/hwdec/hwdec_vt.c'
endif
```

Reply:
Went with the latter.